test-framework/sudo-test/src/lib.rs

```diff
@@ -27,171 +26,131 @@ pub fn is_original_sudo() -> bool {
-    pub fn user_password(&mut self, username: &str, password: &str) -> &mut Self {
-        self.username_to_passwords
-            .insert(username.to_string(), password.to_string());
-        self
-    }
-
-    /// appends content to the `/etc/sudoers` file
-    ///
-    /// NOTE that if this method is not called `/etc/sudoers` will be empty
-    pub fn sudoers(&mut self, sudoers: &str) -> &mut Self {
-        self.sudoers.push_str(sudoers);
-        self.sudoers.push('\n');
-        self
-    }
-
-    const DEFAULT_PAM_D_SUDO: &str = r#"#%PAM-1.0
-
-@include common-auth
-@include common-account
-@include common-session-noninteractive"#;
-
-    /// overwrites the contents of `/etc/pam.d/sudo`
-    ///
-    /// if this method is not called the contents of `/etc/pam.d/sudo` will match the contents of
-    /// the file provided by the `sudo` package
-    pub fn pam_d_sudo(&mut self, pam_conf: &str) -> &mut Self {
-        let pam_d_sudo = self.pam_d_sudo.get_or_insert_with(String::new);
-        pam_d_sudo.push_str(pam_conf);
-        pam_d_sudo.push('\n');
-        self
-    }
-
-    const DEFAULT_SUDOERS_CHOWN: &str = "root:root";
-
-    /// NOTE if unset, defaults to "root:root"
-    pub fn sudoers_chown(&mut self, chown: &str) -> &mut Self {
-        assert!(self.sudoers_chown.is_none(), "sudoers_chown already set");
-        self.sudoers_chown = Some(chown.to_string());
-        self
-    }
-
-    const DEFAULT_SUDOERS_CHMOD: &str = "440";
-
-    /// NOTE if unset, defaults to "440"
-    pub fn sudoers_chmod(&mut self, chmod: &str) -> &mut Self {
-        assert!(self.sudoers_chown.is_none(), "sudoers_chmod already set");
-        self.sudoers_chmod = Some(chmod.to_string());
-        self
-    }
-
-    /// Creates a file at `path` with specified `contents` and permissions
+    /// # Panics
     ///
-    /// NOTE `path` must be absolute
-    pub fn text_file(&mut self, path: &str, chown: &str, chmod: &str, contents: &str) -> &mut Self {
-        assert!(Path::new(path).is_absolute(), "path must be absolute");
-
+    /// - if the `user` has previously been declared
+    pub fn user(&mut self, user: impl Into<User>) -> &mut Self {
+        let user = user.into();
+        let username = &user.name;
         assert!(
-            !self.text_files.contains_key(path),
-            "text file has already been declared"
-        );
-
-        self.text_files.insert(
-            path.to_string(),
-            TextFile {
-                contents: contents.to_string(),
-                chmod: chmod.to_string(),
-                chown: chown.to_string(),
-            },
+            !self.users.contains_key(username),
+            "user {} has already been declared",
+            username
         );
+        self.users.insert(username.to_string(), user);
 
         self
     }
 
     /// builds the test environment
+    ///
+    /// # Panics
+    ///
+    /// - if any specified `user` already exists in the base image
```

Comment:
I'd say "an" instead of "any" but that's me

Reply:
I actually think 'any' makes sense here (since there are multiple).